src/main/java/sally/AddEventCommand.java

```diff
@@ -0,0 +1,89 @@
+package sally;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Represents a command to add an event task.
+ */
+public class AddEventCommand implements Command {
+    private final String input;
+    private Message message;
+
+    /**
+     * Constructs an AddEventCommand object with the given input.
+     *
+     * @param input The input containing task details.
+     */
+    public AddEventCommand(String input) {
+        this.input = input;
+    }
+
+    /**
+     * Executes the AddEventCommand to add an event task.
+     *
+     * @param tasks The TaskList containing tasks.
+     * @param storage The Storage for tasks.
+     * @throws SallyException If there's an issue while executing the command.
+     */
+    @Override
+    public String execute(TaskList tasks, Storage storage) throws SallyException {
+        String[] parts = input.split("/from | /to ");
+        message = new Message();
+
+        if (parts.length < 3) {
+            throw new SallyException(
+                    "OOPS! The description of an event cannot be incomplete, you need a '/from' and '/to'.");
+        }
+
+        String taskDescription = parts[0];
+        String fromDateTimeInput = parts[1];
+        String toDateTimeInput = parts[2];
+
+        if (convertToDateTime(fromDateTimeInput) == null || convertToDateTime(toDateTimeInput) == null) {
+            throw new SallyException("OOPS! The date and time format is invalid.");
+        } else {
+            Event newEvent =
+                    new Event(taskDescription,
+                            convertToDateTime(fromDateTimeInput),
+                            convertToDateTime(toDateTimeInput));
+            tasks.addTask(newEvent);
+            storage.saveTasksToFile(tasks);
+            return message.addMessage(newEvent, tasks.getTaskList().size());
+        }
+    }
+
+    /**
+     * Converts the input string to a LocalDateTime object.
+     *
+     * @param input The input string containing date and time.
+     * @return The converted LocalDateTime object, or null if format is invalid.
+     */
+    private static LocalDateTime convertToDateTime(String input) {
+        String pattern1 = "\\d{4}-\\d{1,2}-\\d{1,2} \\d{4}";
+        String pattern2 = "\\d{4}-\\d{1,2}-\\d{1,2}";
+        String pattern3 = "\\d{4}";
+        Pattern regexPattern1 = Pattern.compile(pattern1);
+        Pattern regexPattern2 = Pattern.compile(pattern2);
+        Pattern regexPattern3 = Pattern.compile(pattern3);
+        Matcher matcher1 = regexPattern1.matcher(input);
+        Matcher matcher2 = regexPattern2.matcher(input);
+        Matcher matcher3 = regexPattern3.matcher(input);
+
```

Comment:
I like how you grouped relevant statements together and added spacing for readability :)